src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Editor/MSBuildXmlAccessor.cs

```diff
@@ -29,12 +29,15 @@ internal class MSBuildXmlAccessor : IProjectXmlAccessor
 
         public async Task<string> GetProjectXmlAsync()
         {
+            var configuredProject = await _unconfiguredProject.GetSuggestedConfiguredProjectAsync().ConfigureAwait(false);
+            if (configuredProject == null)
+            {
+                return string.Empty;
+            }
             using (var access = await _projectLockService.ReadLockAsync())
             {
-                var stringWriter = new EncodingStringWriter(await _unconfiguredProject.GetFileEncodingAsync().ConfigureAwait(true));
-                var projectXml = await access.GetProjectXmlAsync(_unconfiguredProject.FullPath).ConfigureAwait(true);
-                projectXml.Save(stringWriter);
-                return stringWriter.ToString();
+                var projectXml = await access.GetProjectAsync(configuredProject).ConfigureAwait(true);
+                return projectXml.Xml.RawXml;
```

Comment:
Also, `projectXml` might be a misleading name, since it represents the evaluated state of an msbuild file. `evaluatedProject` or just `project` might be more indicative.